tensordict/nn/probabilistic.py

```diff
@@ -453,8 +454,26 @@ def _dist_sample(
             interaction_type = self.default_interaction_type
 
         if interaction_type is InteractionType.MODE:
-            if hasattr(dist, "get_mode"):
-                return dist.get_mode()
+            try:
+                return dist.deterministic_sample
+            except AttributeError:
+                try:
+                    return dist.mean
+                except AttributeError:
+                    raise NotImplementedError(
+                        f"method {type(dist)}.mode is not implemented."
+                    )
+                finally:
+                    warnings.warn(
+                        "deterministic_sample wasn't found when queried. "
+                        f"{type(self).__name__} is falling back on mean instead. "
+                        f"For better code quality and efficiency, make sure to either "
+                        f"provide a distribution with a deterministic_sample attribute or "
+                        f"to change the InteractionMode to the desired value.",
+                        category=UserWarning,
+                    )
+
+        if interaction_type is InteractionType.MODE:
```

Comment:
```suggestion
        elif interaction_type is InteractionType.MODE:
```

Reply:
not necessary, the function either returns or raises just before